### PR TITLE
Updating CI and bumping to 1.17

### PR DESCRIPTION
**Pull request by blockiosaurus**

### CODE REVIEW

File: .github/.env [hunk 1]
```diff
@@ -1,5 +1,7 @@
 CARGO_TERM_COLOR=always
 NODE_VERSION=16.x
 PROGRAMS=["system-extras", "token-extras"]
-RUST_VERSION=1.65.0
-SOLANA_VERSION=1.14.13
+RUST_VERSION=1.70.0
+SOLANA_VERSION=1.17.20
```
Comment (danenbm) [inline .github/.env:5]: So is it time for us to start making all our CI use 1.17.20 probably?

File: .github/file-filters.yml [hunk 2]
```diff
@@ -27,6 +27,8 @@ client_common: &client_common
   - *programs
   - ".github/workflows/test-js.yml"
```
Comment (danenbm) [inline .github/file-filters.yml:28]: i think this should be `test-js-client.yml`?

**Review by danenbm — CHANGES_REQUESTED** [review]

File: .github/.env (hunk 1 above)
Comment (blockiosaurus) [reply .github/.env:5]: Probably, yeah

**Review by danenbm — APPROVED** [review]
LGTM!

**Pull request merged**